Sachin miniproject

SachinM123 opened the pull request:
Made a Talon motor go (change speed forward and backward) based on right joystick X movement.

SachinM123 committed:
Add initial Xbox controller motor control project

Set up a new WPILib Java project for controlling a motor using the right stick X direction of an Xbox controller. Includes Gradle build files, VSCode configuration, WPILib preferences, vendor dependencies, and source code for robot, container, motor subsystem, and commands.

SachinM123 committed:
Set MoveMotor as default command for MotorSubsystem

MoveMotor is now assigned as the default command for MotorSubsystem using the specified controller. The command's isFinished method is overridden to always return false, ensuring continuous operation. Unused code and redundant trigger bindings were removed for clarity.

4682cybears committed:
ids

4682cybears committed:
z

z

SachinM123 committed:
Remove duplicate controller and update bindings

Eliminated redundant CommandXboxController instance and updated the command binding to use the existing controller reference for improved clarity and maintainability.

SachinM123 committed:
Refactor MoveMotor to use DoubleSupplier input

Renamed MoveMotor to MoveMotorCommand and updated its constructor to accept a DoubleSupplier for joystick input instead of a CommandXboxController. Updated usages in RobotContainer and Autos to pass the right stick X value via lambda. Improved MotorSubsystem to stop the motor when speed is zero.

SachinM123 committed:
changes for test

SachinM123 committed:
pr comments

SachinM123 committed:
Update RobotContainer.java

SachinM123 committed:
a

a

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/deploy/example.txt:
```diff
@@ -0,0 +1,3 @@
+Files placed in this directory will be deployed to the RoboRIO into the
```
remove unused file.

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,33 @@
+    public static final int talonFXCANID = 4;
+  }
+
+  private Constants() {
```
remove this unused constructor. 
Also, *never* throw an error in robot code. We don't want the roborio to crash mid-match. We always need to find a way to handle exceptions gracefully without crashing.

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,60 @@
+   * @return the command to run in autonomous
+   */
+  public Command getAutonomousCommand() {
+    return Commands.print("No autonomous command configured");
```
we usually just return a new InstantCommand() here.

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/commands/Autos.java:
```diff
@@ -0,0 +1,40 @@
+// ************************************************************
```
remove this unused file.

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/commands/MoveMotorCommand.java:
```diff
@@ -0,0 +1,74 @@
+  /**
+   * Initializes the command. Called when the command is initially scheduled.
+   */
+  @Override
```
if you don't have any initialization code, you can remove this method.

josieammer replied:
you still need to address this comment

SachinM123 replied:
Removed method

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/commands/MoveMotorDefaultCommand.java:
```diff
@@ -0,0 +1,74 @@
+// ************************************************************
```
Since this is used as a DefaultCommand, it should have "DefaultCommand" in the file name.

josieammer replied:
you still need to address this comment

SachinM123 replied:
Done

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,67 @@
+ */
+public class MotorSubsystem extends SubsystemBase {
+
+  private final TalonFX motor = new TalonFX(OperatorConstants.talonFXCANID);
```
It makes this sussystem more reusable if you take in the CANID as a parameter to the constructor, rather than hard-coding it here.

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,67 @@
+   * @param speed Motor speed from -1.0 to 1.0
+   */
+  public void setMotorSpeed(double speed) {
+    speed = Math.max(-1.0, Math.min(1.0, speed));
```
you should use MathUtil.clamp() here, rather than doing the min/max math yourself.

josieammer replied:
You still need to address this comment.

SachinM123 replied:
Done

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,67 @@
+   * Creates an example command.
+   * @return A command that runs once
+   */
+  public Command exampleMethodCommand() {
```
remove unused method.

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,67 @@
+   * Queries a boolean state of the subsystem.
+   * @return The current state
+   */
+  public boolean exampleCondition() {
```
remove unused method.

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,67 @@
+    return false;
+  }
+
+  @Override
```
remove unused method.

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,67 @@
+    // This method will be called once per scheduler run
+  }
+
+  @Override
```
remove unused method

josieammer commented on vendordeps/WPILibNewCommands.json:
```diff
@@ -0,0 +1,38 @@
+{
```
you seem to have a duplicate vendordeps directory out here at the wrong level of the hierarchy. Keep the one in your directory and delete this one.

josieammer replied:
you still need to delete this file.

SachinM123 replied:
removing this file makes the project no longer build; giving this error: Starting a Gradle Daemon, 2 busy and 5 incompatible and 3 stopped Daemons could not be reused, use --status for details

> Task :compileJava
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\commands\MoveMotorDefaultCommand.java:13: error: package edu.wpi.first.wpilibj2.command does not exist
import edu.wpi.first.wpilibj2.command.Command;
                                     ^
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\commands\MoveMotorDefaultCommand.java:21: error: cannot find symbol
public class MoveMotorDefaultCommand extends Command {
                                             ^
  symbol: class Command
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\subsystems\MotorSubsystem.java:18: error: package edu.wpi.first.wpilibj2.command does not exist
import edu.wpi.first.wpilibj2.command.Command;
                                     ^
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\subsystems\MotorSubsystem.java:19: error: package edu.wpi.first.wpilibj2.command does not exist
import edu.wpi.first.wpilibj2.command.SubsystemBase;
                                     ^
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\subsystems\MotorSubsystem.java:26: error: cannot find symbol
public class MotorSubsystem extends SubsystemBase {
                                    ^
  symbol: class SubsystemBase
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\Robot.java:8: error: package edu.wpi.first.wpilibj2.command does not exist
import edu.wpi.first.wpilibj2.command.Command;
                                     ^
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\Robot.java:9: error: package edu.wpi.first.wpilibj2.command does not exist
import edu.wpi.first.wpilibj2.command.CommandScheduler;
                                     ^
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\Robot.java:17: error: cannot find symbol
  private Command m_autonomousCommand;
          ^
  symbol:   class Command
  location: class Robot
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:16: error: package edu.wpi.first.wpilibj2.command does not exist
import edu.wpi.first.wpilibj2.command.Command;
                                     ^
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:17: error: package edu.wpi.first.wpilibj2.command does not exist
import edu.wpi.first.wpilibj2.command.Commands;
                                     ^
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:18: error: package edu.wpi.first.wpilibj2.command does not exist
import edu.wpi.first.wpilibj2.command.InstantCommand;
                                     ^
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:19: error: package edu.wpi.first.wpilibj2.command.button does not exist
import edu.wpi.first.wpilibj2.command.button.CommandXboxController;
                                            ^
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:30: error: cannot find symbol  
  private final CommandXboxController driverController =
                ^
  symbol:   class CommandXboxController
  location: class RobotContainer
C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:58: error: cannot find symbol  
  public Command getAutonomousCommand() {
         ^
  symbol:   class Command
  location: class RobotContainer
14 errors

> Task :compileJava FAILED
Compilation Error!
GradleRIO detected this build failed due to a Compile Error (compileJava).
Check that all your files are saved, then scroll up in this log for more information.
[Incubating] Problems report is available at: file:///C:/Users/Sachi/Documents/GitHub/FirstAge2026/camp/sachin-Xbox-Controller-Right-Stick-X-direction/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler output below.
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\commands\MoveMotorDefaultCommand.java:21: error: cannot find symbol
  public class MoveMotorDefaultCommand extends Command {
                                               ^
    symbol: class Command
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\subsystems\MotorSubsystem.java:26: error: cannot find symbol
  public class MotorSubsystem extends SubsystemBase {
                                      ^
    symbol: class SubsystemBase
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\commands\MoveMotorDefaultCommand.java:13: error: package edu.wpi.first.wpilibj2.command does not exist
  import edu.wpi.first.wpilibj2.command.Command;
                                       ^
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\subsystems\MotorSubsystem.java:18: error: package edu.wpi.first.wpilibj2.command does not exist
  import edu.wpi.first.wpilibj2.command.Command;
                                       ^
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\subsystems\MotorSubsystem.java:19: error: package edu.wpi.first.wpilibj2.command does not exist
  import edu.wpi.first.wpilibj2.command.SubsystemBase;
                                       ^
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\Robot.java:8: error: package edu.wpi.first.wpilibj2.command does not exist
  import edu.wpi.first.wpilibj2.command.Command;
                                       ^
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\Robot.java:9: error: package edu.wpi.first.wpilibj2.command does not exist
  import edu.wpi.first.wpilibj2.command.CommandScheduler;
                                       ^
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:16: error: package edu.wpi.first.wpilibj2.command does not exist
  import edu.wpi.first.wpilibj2.command.Command;
                                       ^
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:17: error: package edu.wpi.first.wpilibj2.command does not exist
  import edu.wpi.first.wpilibj2.command.Commands;
                                       ^
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:18: error: package edu.wpi.first.wpilibj2.command does not exist
  import edu.wpi.first.wpilibj2.command.InstantCommand;
                                       ^
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:19: error: package edu.wpi.first.wpilibj2.command.button does not exist
  import edu.wpi.first.wpilibj2.command.button.CommandXboxController;
                                              ^
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\Robot.java:17: error: cannot find symbol
    private Command m_autonomousCommand;
            ^
    symbol:   class Command
    location: class Robot
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:30: error: cannot find symbol
    private final CommandXboxController driverController =
                  ^
    symbol:   class CommandXboxController
    location: class RobotContainer
  C:\Users\Sachi\Documents\GitHub\FirstAge2026\camp\sachin-Xbox-Controller-Right-Stick-X-direction\src\main\java\frc\robot\RobotContainer.java:58: error: cannot find symbol
    public Command getAutonomousCommand() {
           ^
    symbol:   class Command
    location: class RobotContainer
  14 errors
  
  I'm unsure why it does this; should I remove the file anyway?

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,67 @@
+   * Constructs a MotorSubsystem and configures the motor.
+   */
+  public MotorSubsystem() {
+    motor.setNeutralMode(NeutralModeValue.Brake);
```
The way your are applying the config to this motor has been deprecated for 2026. It's only still working because we are still using the 2025 library for our mini projects. See Asher's code for how to properly apply the config to Talon Motors. 
https://github.com/Team4682CyBears/FirstAge2026/blob/39b1daea395dd25361c4dd8c1b27c197953ed7e2/camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java#L48

josieammer reviewed (changes requested):
Good start! Just a few things to clean up.

SachinM123 committed:
Refactor motor config and remove unused files

Moved motor configuration logic into a dedicated method in MotorSubsystem and updated usage of constants. Deleted unused Autos.java, example.txt, and Phoenix5 vendordep files. Simplified autonomous command and removed unnecessary example methods.

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,79 @@
+    motor.set(speed);
+  }
+  
+  /**
```
remove this unused comment

SachinM123 replied:
Done

josieammer commented on camp/sachin-Xbox-Controller-Right-Stick-X-direction/src/main/java/frc/robot/subsystems/MotorSubsystem.java:
```diff
@@ -0,0 +1,79 @@
+   */
+ 
+
+  /**
```
remove this unused comment

SachinM123 replied:
Done

josieammer reviewed (changes requested):
Good progress. There are still a few comments that you didn't yet address. 

Also, when you address a comment, please reply to every comment. Even if you just say 'done' or 'fixed', it helps me go faster when I review it again.

SachinM123 committed:
Rename MoveMotorCommand to MoveMotorDefaultCommand

Renamed MoveMotorCommand to MoveMotorDefaultCommand for clarity and updated references in RobotContainer. Also replaced manual clamping in MotorSubsystem with MathUtil.clamp for improved readability and reliability.

josieammer committed:
remove unnecessary vendordeps

josieammer reviewed (approved):
Looks good. I've removed the vendordeps for you. It builds for me. I think you were trying to delete the one inside your project folder rather than the duplicate one at the top level.
